Replicator/DBAccess.cc

```diff
@@ -82,28 +81,30 @@ namespace litecore { namespace repl {
 
 
     void DBAccess::close() {
+        if (_closed.test_and_set()) {
+            return;
+        }
         _timer.stop();
-        use([&](C4Database *&db) {
+        use([this](C4Database *&db) {
             // Any use of the class after this will result in a crash that 
             // should be easily identifiable, so forgo asserting if the pointer 
             // is null in other areas.
             c4db_release(db);
             db = nullptr;
+            this->_sentry = &DBAccess::AssertDBOpen;
```

Comment:
We don't do c4db_release(db) in the current master branch. Can you double check if it's correct to do the release here?

Reply:
In 2.8, DBAccess is access_lock<C4Database*>, whereas in 3.0, it is access_lock<Retained<C4Database>>. Therefore, we need to manually do c4db_release in 2.8